.github/workflows/build_test.yaml

```diff
@@ -49,4 +49,21 @@ jobs:
       uses: codecov/codecov-action@v3
       env:
         CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
-      
\ No newline at end of file
+      
+  release:
+    runs-on: ubuntu-latest
+    needs: test
+    if: github.event_name == 'create'
```

Comment:
I think the `on: tag` mechanism is better than this

Also we need to build the file to make the release, there is no build step here. 

Are you sure of what you're doing here? It is ok if you have never used Github Actions before - but let's not make PRs on guesswork.

Reply:
Fair enough. I have some idea of what I think could be happening but it would be better if I could know how to run this and see if it's working properly?

Reply:
I am going to try this on a personal repo and come back once I figure this out. This is good learning nonetheless.